url64d634b8f0_agdturner_generic_tgz-pJ8-uk_ac_leeds_ccg_andyt_generic_visualisation_charts_Generic_BarChartJ8/src/uk/ac/leeds/ccg/andyt/generic/memory/Generic_OutOfMemoryErrorHandler.java

```diff
@@ -0,0 +1,466 @@
+                        handleOutOfMemoryError);
+            } else {
+                throw a_OutOfMemoryError;
+            }
+        }
+    }
+
+    /**
+     * @param a_String
+     * @param b_String
+     * @param handleOutOfMemoryError
+     * @return String which is a_String appended to the end of b_String
+     */
+    public String initString(
+            String a_String,
+            String b_String,
+            boolean handleOutOfMemoryError) {
+        try {
+            String result = a_String + b_String;
+            tryToEnsureThereIsEnoughMemoryToContinue();
+            return result;
+        } catch (OutOfMemoryError a_OutOfMemoryError) {
+            if (handleOutOfMemoryError) {
+                clear_MemoryReserve();
+                if (swapToFile_DataAny()){
+                    init_MemoryReserve(
+                       HandleOutOfMemoryErrorFalse);
+                } else {
+                    throw a_OutOfMemoryError;
+                }
+                return initString(
+                        a_String,
+                        b_String,
+                        handleOutOfMemoryError);
+            } else {
+                throw a_OutOfMemoryError;
+            }
+        }
+    }
+
+    /**
+     * For initialising a File from String _String.
+     * @param a_String
+     * @param handleOutOfMemoryError
+     * @return 
+     * @throws java.io.IOException 
+     */
+    public File initFile(
+            String a_String,
+            boolean handleOutOfMemoryError)
+            throws IOException {
+        try {
+            File result = new File(a_String);
+            result.getParentFile().mkdirs();
+            result.createNewFile();
+            tryToEnsureThereIsEnoughMemoryToContinue();
+            return result;
+        } catch (OutOfMemoryError a_OutOfMemoryError) {
+            if (handleOutOfMemoryError) {
+                clear_MemoryReserve();
+                if (swapToFile_DataAny()){
+                    init_MemoryReserve(
+                       HandleOutOfMemoryErrorFalse);
+                } else {
+                    throw a_OutOfMemoryError;
+                }
+                init_MemoryReserve(
+                        handleOutOfMemoryError);
+                return initFile(
+                        a_String,
+                        handleOutOfMemoryError);
+            } else {
+                throw a_OutOfMemoryError;
+            }
+        }
+    }
+
+    /**
+     * For initialising a File from File _File and String _String.
+     * @param a_File
+     * @param a_String
+     * @param handleOutOfMemoryError
+     * @return 
+     * @throws java.io.IOException 
+     */
+    public File initFile(
+            File a_File,
+            String a_String,
+            boolean handleOutOfMemoryError)
+            throws IOException {
+        try {
+            File result = new File(a_File, a_String);
+            a_File.mkdirs();
+            result.createNewFile();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

`File.createNewFile` returns `false` if the named file already exists and an empty file is not created. Check the value returned by `createNewFile()`. Handle when the method returns `false`. Otherwise, you might silently overwrite an existing file which could lead to data loss. See the following diffs on GitHub that make this code change: [diff1](https://github.com/lingochamp/FileDownloader/commit/65069c18a4b1b4cfa758173ef49d69c6897fc77a#diff-2b6ad96df9bf9828c22f961e9e9a5b933ab2df7b2fb0f12d62c044309e1e2842L331), [diff2](https://github.com/m-szalik/javamail/commit/62b1cb45e32a998efbe6cfa09b06752869f590ae#diff-440ef5498fc2c802274cf136e7bd7d682bbc10c5976afb5af718984e24deced4L45)

Learn more: [File.createNewFile in the Java Platform, Standard Edition 8 API Specification](https://docs.oracle.com/javase/8/docs/api/java/io/File.html#createNewFile--)